### PR TITLE
Prod 1483, 1485, 1487 Topcoder University - landing, details, course outline

### PR DESCRIPTION
**Pull request by vas3a**
There's no ticket for this one, it's just me running tslint fixes over the previous PRs and refactoring the data model in all the components. It got easier for me to do it like so since I had a clearer view of the necessary data.

So basically this contains all work from PRs #429, #430 and #431 + tslint and data refectoring.

### CODE REVIEW

**Review by brooketopcoder — CHANGES_REQUESTED** [review]
This looks great! We just need to fix that typo w/the link variable name and we should be good.

File: src-ts/lib/learn-provider/providers/my-certification-progress-provider.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+const myCertifications: Array<LearnMyCertification> = [...myCertificationsJSON.certifications] as Array<LearnMyCertification>
+
+export interface MyCertificationProgressProviderData {
+    loading: boolean
+    progress?: LearnMyCertificationProgress
+    ready: boolean
+}
+
+export const useMyCertificationProgress: (certification?: string) => MyCertificationProgressProviderData = (certification?: string): MyCertificationProgressProviderData => {
+    const [state, setState]: [MyCertificationProgressProviderData, Dispatch<SetStateAction<MyCertificationProgressProviderData>>] = useState<MyCertificationProgressProviderData>({
```
[the remark this answers is not in the document]
Comment (vas3a) [reply src-ts/lib/learn-provider/providers/my-certification-progress-provider.tsx]: Got it! Ok, I'll account for this from now on!

[the remark this answers is not in the document]
Comment (brooketopcoder) [reply src-ts/lib/learn-provider/providers/my-certification-progress-provider.tsx]: thanks!

File: src-ts/tools/learn/services/learn-provider/providers/certifications.provider.tsx [hunk 2]
```diff
@@ -1,7 +1,8 @@
 import { Dispatch, SetStateAction, useCallback, useEffect, useMemo, useState } from 'react'
 
-import certificationsJSON from '../../../assets/data/certifications.json'
-import { LearnCertification } from '../models'
+import certificationsJSON from '../../../../../assets/data/certifications.json'
```
Comment (brooketopcoder) [inline src-ts/tools/learn/services/learn-provider/providers/certifications.provider.tsx:3]: let's move this assets folder into the learn directory as well

**Review by brooketopcoder — CHANGES_REQUESTED** [review]
This looks fantastic--thanks! Just one more super minor change...

**Review by testflyjets — APPROVED** [review]

**Pull request merged**